modules/vba_documents/lib/vba_documents/multipart_parser.rb

```diff
@@ -1,32 +1,29 @@
 # frozen_string_literal: true
 
 require 'vba_documents/upload_error'
+require 'common/file_helpers'
+require 'common/virus_scan'
 
 module VBADocuments
   class MultipartParser
     LINE_BREAK = "\r\n"
-    CARRIAGE_RETURN = "\r"
     BASE64_PREFIX = 'data:multipart/form-data;base64,'
 
-    def self.parse(infile)
-      if base64_encoded?(infile)
-        create_file_from_base64(infile)
-      else
-        parse_file(infile)
-      end
+    def self.parse(file_path)
+      file_path = decode_base64_file(file_path) if base64_encoded_file?(file_path)
+
+      validate_virus_free(file_path) if Flipper.enabled?(:vba_documents_virus_scan)
+
+      parse_file(file_path)
```

Comment:
Previously the method that decoded a Base64 file called `parse_file` directly, but this made it easy to lose the path of execution when reading the class. Now `parse_file` is only called from within the `parse` method.